frontend/src/components/dashboard/settings/api-key/ApiKey.tsx

```diff
@@ -1,155 +1,135 @@
+    const apiKey = apiKeys.find((k) => k.id === keyId)
+    if (!apiKey) return
+    modalContext.setModalContent(
+      <ConfirmModal
+        title={`Are you sure you want to delete API key "${apiKey.label}"?`}
+        subtitle="Deleting your API key is immediate and irreversible. To ensure a smooth API key rotation, please replace this key with a new API key before deleting."
+        onConfirm={() => {
+          void deleteApiKey(keyId).then(() => removeApiKey(keyId))
+        }}
+        onCancel={() => modalContext.close()}
+        buttonText="Confirm delete"
+        buttonIcon="bx-trash"
+        cancelText="Cancel"
+        destructive
+      />
+    )
   }
 
-  return (
+  return isPageLoading ? (
+    <div className={styles.loader}>
+      <i className="bx bx-loader-alt bx-spin" />
+    </div>
+  ) : (
     <>
-      <StepHeader title="API Key">
-        <p className={styles.helpText}>
-          After generating your API key, please make a copy of it immediately as
-          it will only be shown once. Upon leaving or refreshing this page, the
-          key will be hidden.
-        </p>
-      </StepHeader>
-      <TextInputWithButton
-        value={apiKey}
-        onChange={() => {
-          return
-        }}
-        onClick={onButtonClick}
-        className={buttonClass}
-        textRef={apiKeyRef}
-        errorMessage={errorMsg}
-        buttonLabel={
-          <>
-            {buttonLabel} API key
-            <i className={cx('bx', buttonIcon)} />
-          </>
-        }
-      />
+      <div className={styles.header}>
+        <h3>API Key</h3>
+        <PrimaryButton
+          className={styles.blueButton}
+          onClick={openGenerateModal}
+        >
+          <i className="bx bx-plus"></i>
+          Generate API Key
+        </PrimaryButton>
+      </div>
+      <p className={styles.helpText}>
+        You can create an API key to access our programmatic email and
+        programmatic SMS APIs. For more information, go to our{' '}
+        <OutboundLink
+          eventLabel="https://guide.postman.gov.sg/developer-guide/api-doc"
+          to="https://guide.postman.gov.sg/developer-guide/api-doc"
+          target="_blank"
+        >
+          API Guide
+        </OutboundLink>
+        .
+      </p>
+      {apiKeys.length > 0 && (
+        <table className={styles.apiKeyTable}>
+          <thead>
+            <tr>
+              <th className="lg">Key Label</th>
+              <th className="sm">Last 5 Digits</th>
+              <th className="sm"></th>
+            </tr>
+          </thead>
+          <tbody>
+            {apiKeys.map((k) => (
+              <tr key={k.id}>
+                <td className="lg">{k.label}</td>
+                <td className="sm">••••• {k.last_five}</td>
+                <td className={cx('sm', styles.buttonContainer)}>
+                  {/* <button>
+                <i className={cx('bx bx-pencil', styles.pencil)} />
+              </button> */}
```

Comment:
**[question]** why is this commented out and not just deleted ah?

Reply:
cos we're using it in the next iteration so I'm leaving there haha. During compilation these will be removed from the final code anyway

Reply:
> cos we're using it in the next iteration so I'm leaving there haha. During compilation these will be removed from the final code anyway

coolz sgtm

let's coordinate with @kendra-wong to update the guide before merging into `master` (am aware this one is just merging into `feat/generate-api-key`)

Reply:
I'll update the guide